bug(build): support spdlog 1.16 and fmt 12 on Arch Linux

```
❯ pacman -Q spdlog fmt clang libc++
spdlog 1.16.0-1
fmt 12.1.0-1
clang 21.1.4-2
libc++ 21.1.4-1
```
```
❯ xmake f --toolchain=clang --dev=y --cxxflags="-stdlib=libc++" --ldflags="-stdlib=libc++" -m debug
checking for platform ... linux
checking for architecture ... x86_64
checking for llvm ... no
```
```
❯ xmake f -c
checking for platform ... linux
checking for architecture ... x86_64
checking for llvm ... no
```
```
❯ xmake run unit_tests
...
[ 58%]: cache compiling.release src/Server/Document.cpp
error: include/Support/Logging.h:50:20: error: no matching function for call to ‘log(spdlog::source_loc&, spd
   50 |         spdlog::log(loc, level, fmt, std::forward<Args>(args)...);
      |         ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
include/Support/Logging.h:50:20: note: there are 4 candidates
In file included from include/Support/Logging.h:4:
/usr/include/spdlog/spdlog.h:149:13: note: candidate 1: ‘void spdlog::log(source_loc, level::level_enum, fmt::v12::format_string<T ...>, Args&& ...) [with Args = {const char*}; fmt::v12::format_string<T ...> = fmt::v12::fstring<const char*>]’
  149 | inline void log(source_loc source,
      |             ^~~
/usr/include/spdlog/spdlog.h:151:42: note: no known conversion for argument 3 from ‘std::format_string<const char*>’ {aka ‘std::basic_format_string<char, const char*>’} to ‘fmt::v12::format_string<const char*>’ {aka ‘fmt::v12::fstring<const char*>’}
  151 |                 format_string_t<Args...> fmt,
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~^~~
/usr/include/spdlog/spdlog.h:157:13: note: candidate 2: ‘template<class ... Args> void spdlog::log(level::level_enum, fmt::v12::format_string<T ...>, Args&& ...)’
  157 | inline void log(level::level_enum lvl, format_string_t<Args...> fmt, Args &&...args) {
      |             ^~~
/usr/include/spdlog/spdlog.h:157:13: note: template argument deduction/substitution failed:
include/Support/Logging.h:50:21: note:   cannot convert ‘loc’ (type ‘spdlog::source_loc’) to type ‘spdlog::level::level_enum’
   50 |         spdlog::log(loc, level, fmt, std::forward<Args>(args)...);
  > in src/Async/Network.cpp
```

--- a/xmake.lua
+++ b/xmake.lua
@@ -40,7 +40,8 @@ if has_config("release") then
 end
 
 add_defines("TOML_EXCEPTIONS=0")
-add_requires(libuv_require, "spdlog[header_only=n,std_format,noexcept]" ,"toml++", "croaring", "flatbuffers")
+add_requires("spdlog", {system=false, version="1.15.3", configs = {header_only = false, std_format = true, noexcept = true}})
+add_requires(libuv_require, "toml++", "croaring", "flatbuffers")
 add_requires("clice-llvm", {alias = "llvm"})
 
 add_rules("mode.release", "mode.debug", "mode.releasedbg")

